nominatim/tools/special_phrases/sp_loader.py

```diff
@@ -0,0 +1,16 @@
+"""
+    Module containing the SPLoader class.
+"""
+from abc import ABC, abstractmethod
+
+class SPLoader(ABC):
+    """
+        Base class for special phrases loaders.
+        Handle the loading of special phrases from external sources.
+    """
+    def __iter__(self):
+        return self
+
+    @abstractmethod
+    def __next__(self):
+        pass
```

Comment:
As Python is not a strictly typed language, you don't need a common base class for your loader class. You can rely on duck typing instead and simply make sure that they implement all functions needed for the loader. So in this case, the SP*Loader classes can simply inherit from `collections.abc.Iterator` to get the `__iter__` implementation. And this SPLoader class can be dropped.